src/liballoc/vec.rs

```diff
@@ -171,7 +171,7 @@ use raw_vec::RawVec;
 ///
 /// # Slicing
 ///
-/// A `Vec` can be mutable. Slices, on the other hand, are read-only objects.
+/// A slice is a reference to a part of the vector.
```

Comment:
(1) The new text could be easily read as "a slice is [always] a reference to a part of [a] vector". Can you think of a simple way to make it a bit clearer that slices can reference other stuff (not just vector contents) as well? If not, that's fine with me as well, and we can keep the current text.

(2) I'm not sure what the original intent was, but I think that "mutable" could have referred to "growable". Do you think it would make sense to have that distinction here?